src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -162,6 +163,16 @@ public class JsonMessageFormatter : IJsonRpcAsyncMessageTextFormatter, IJsonRpcF
         /// </summary>
         private JsonRpcMessage? deserializingMessage;
 
+        /// <summary>
+        /// Whether <see cref="EnforceFormatterIsInitialized"/> has been executed.
+        /// </summary>
+        private bool formatterInitializationChecked;
+
+        /// <summary>
+        /// Object used to lock when running <see cref="EnforceFormatterIsInitialized"/>.
+        /// </summary>
+        private object formatterInitializationLock = new();
```

Comment:
Studies have shown more than one lock field on a class greatly increases odds of threading bugs. So rather than dedicate this field to just locking one particular thing, I'd like to consider it _the_ object to lock for anything the formatter requires. To that end, can you please rename it to `syncObject` to match the naming convention used elsewhere?
Also please add the `readonly` modifier. This defends against thread-safety bugs when/if this field were ever reassigned (which I have seen as real bugs in the wild before). StyleCop will require that you move the field up at that point to be amongst the other `readonly` fields.